app/models/publish_services_metadata_event.rb

```diff
@@ -0,0 +1,29 @@
+class PublishServicesMetadataEvent < Event
+  
+  data_attributes :event_id, :cert_config
+  before_create :populate_data_attributes
+  after_create :upload
+  has_one_attached :document
+
+  def populate_data_attributes
+    assign_attributes(attributes_to_apply)
+  end
+
+  def attributes_to_apply
```

Comment:
I think this method should be inlined into `populate_data_attributes` as the name overlaps with `AggregatedEvent#attributes_to_apply` which is used for a different purpose: applying the attributes from an event to an aggregate.